File: AtlasWeb/Core/Models/LogisticReportViewModel.cs
using System;
using System.Collections.Generic;
using TTDataDictionary;

namespace Core.Models
{
    public class LogisticReportViewModel
    {
        public List<Report> Reports
        {
            get;
            set;
        }
    }

    public class Report
    {
        public Guid id
        {
            get;
            set;
        }

        public string name
        {
            get;
            set;
        }

        public string caption
        {
            get;
            set;
        }

        public List<Report> items
        {
            get;
            set;
        }
    }

    public class SubStoreModel
    {
        public List<SubStoreItem> Stores
        {
            get;
            set;
        }
    }

    public class SubStoreItem
    {
        public string id
        {
            get;
            set;
        }

        public string name
        {
            get;
            set;
        }
    }
}

File: AtlasWeb/Core/wwwroot/app/Logistic/Models/LogisticReportViewModel.ts
import { BaseModel } from 'Fw/Models/BaseModel';
import { Type } from 'NebulaClient/ClassTransformer';
import { Guid } from 'NebulaClient/Mscorlib/Guid';

export class LogisticReportViewModel extends BaseModel {
    public Reports: Array<Report>;

}

export class Report {
    @Type(() => Guid)
    id: Guid;
    name: string;
    caption: string;
    items?: Report[];
}

export class SubStoreModel {
    public Stores: SubStoreItem[];
}
export class SubStoreItem {
    public id: string;
    public name: string;
}